kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveGlobalStore.java

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * The {@code ResponsiveGlobalStore}
+ */
+public class ResponsiveGlobalStore implements KeyValueStore<Bytes, byte[]> {
+
+  private final CassandraClient client;
+  private final String name;
+  private final String tableName;
+  private final RemoteMonitor initRemote;
+  private final Position position;
+
+  private boolean open;
+  private int partition;
+  private StateStoreContext context;
+
+  public ResponsiveGlobalStore(
+      final CassandraClient client,
+      final String name,
+      final RemoteMonitor initRemote
+  ) {
+    this.client = client;
+    this.name = name;
+    this.tableName = '"' + name + '"';
+    this.initRemote = initRemote;
+    this.position = Position.emptyPosition();
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public void init(final ProcessorContext context, final StateStore root) {
+    if (context instanceof StateStoreContext) {
+      init((StateStoreContext) context, root);
+    } else {
+      throw new UnsupportedOperationException(
+          "Use ResponsiveStore#init(StateStoreContext, StateStore) instead."
+      );
+    }
+  }
+
+  @Override
+  public void init(final StateStoreContext context, final StateStore root) {
+    try {
+      this.context = context;
+      // this is bad, but the assumption is that global tables are small
+      // and can fit in a single partition - all writers will write using
+      // the same partition key. we should consider using ALLOW FILTERING
+      // instead. that would make gets/puts more efficient and the queries
+      // to be more evently distributed and take the hit only on range
+      // queries
+      partition = 0;
+      client.createDataTable(tableName);
+      initRemote.await(Duration.ofSeconds(60));
+
+      client.prepareStatements(tableName);
+      client.initializeOffset(tableName, partition);
+
+      open = true;
+
+      context.register(root, (key, value) -> {
+        if (value == null) {
+          delete(new Bytes(key));
+        } else {
+          put(new Bytes(key), value);
+        }
+      });
+    } catch (InterruptedException | TimeoutException e) {
+      throw new ProcessorStateException("Failed to initialize store.", e);
+    }
+  }
+
+  @Override
+  public boolean isOpen() {
+    return open;
+  }
+
+  @Override
+  public boolean persistent() {
+    return false;
+  }
+
+  @Override
+  public void put(final Bytes key, final byte[] value) {
+    client.execute(client.insertData(tableName, partition, key, value));
+    StoreQueryUtils.updatePosition(position, context);
+  }
+
+  @Override
+  public byte[] putIfAbsent(final Bytes key, final byte[] value) {
+    final byte[] old = get(key);
+    if (old == null) {
+      put(key, value);
+    }
+    return old;
+  }
+
+  @Override
+  public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+    entries.forEach(kv -> put(kv.key, kv.value));
```

Comment:
See comment about batching above...technically `putAll` itself might never be invoked unless we do it ourselves but we may as well put the batching logic here and then just call it above